.github/workflows/sdk.yml

```diff
@@ -0,0 +1,65 @@
+on:
+  workflow_call:
+    inputs:
+      runsOn:
+        required: true
+        type: string
+      target:
+        required: true
+        type: string
+
+env:
+  DOTNET_CLI_TELEMETRY_OPTOUT: 1
+  DOTNET_NOLOGO: 1
+
+jobs:
+  build:
+    runs-on: ${{ inputs.runsOn }}
+    timeout-minutes: 30
+    env:
+      submodules: modules/sentry-java
+      sdk: Android
+    steps:
+      # Tag: 0.9.1
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@a40b8845c0683271d9f53dfcb887a7e181d3918b
+        with:
+          access_token: ${{ github.token }}
+      - uses: actions/checkout@v2
+      - name: Get submodules
+        shell: bash
+        run: |
+          if [[ "${{ inputs.target }}" == "Android" ]]; then
+            submodules="modules/sentry-java"
+          elif [[ "${{ inputs.target }}" == "iOS" ]]; then
+            submodules="modules/sentry-cocoa"
+          else
+            submodules="modules/sentry-native"
+          fi
+          ./scripts/init-submodules.sh src/sentry-dotnet $submodules
+          git submodule status > submodules-status
+          cat submodules-status
+      - name: Restore from cache
+        uses: actions/cache@v2
+        with:
+          # Note: native SDKs are cached and only built if the respective 'package-dev/Plugins/' directories are empty.
+          # Output changes only depending on the git sha of the submodules
+          # hash of package/package.json for cache busting on release builds (version bump)
+          path: |
+            package-dev/Plugins
+            modules/sentry-java/sentry-android-ndk/build/intermediates/merged_native_libs/release/out/lib
+          key: sdk=${{ inputs.target }}-modules=${{ hashFiles('submodules-status') }}-package=${{ hashFiles('package/package.json') }}
+      - name: Set up JDK 11
+        if: ${{ inputs.target }} == "Android"
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Build
+        run: dotnet msbuild /t:Build${{ inputs.target }}SDK /p:Configuration=Release /p:OutDir=other src/Sentry.Unity
```

Comment:
Those targets depend on Build which will restore all submodules. In CI we had it restore in the CI step so that caching would have the folders to restore files into

https://github.com/getsentry/sentry-unity/blob/9b0b81ae436d99968aa718a1c5d47e1898a009ea/before.Sentry.Unity.sln.targets#L1-L11

Reply:
I'll check but it doesn't seem to happen, e.g. windows build:

```
Run dotnet msbuild /t:BuildWindowsSDK /p:Configuration=Release /p:OutDir=other src/Sentry.Unity
  dotnet msbuild /t:BuildWindowsSDK /p:Configuration=Release /p:OutDir=other src/Sentry.Unity
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    DOTNET_CLI_TELEMETRY_OPTOUT: 1
    DOTNET_NOLOGO: 1
    submodules: modules/sentry-java
    sdk: Android
    
    JAVA_HOME: C:\hostedtoolcache\windows\Java_Adopt_jdk\11.0.14-9\x64
Microsoft (R) Build Engine version 16.11.[2](https://github.com/getsentry/sentry-unity/runs/5077701645?check_suite_focus=true#step:7:2)+f[3](https://github.com/getsentry/sentry-unity/runs/5077701645?check_suite_focus=true#step:7:3)22596[4](https://github.com/getsentry/sentry-unity/runs/5077701645?check_suite_focus=true#step:7:4)2 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  -- Building for: Visual Studio 1[6](https://github.com/getsentry/sentry-unity/runs/5077701645?check_suite_focus=true#step:7:6) 2019
  -- The C compiler identification is MSVC 19.29.30139.0
  -- The CXX compiler identification is MSVC 19.29.30139.0
  -- The ASM compiler identification is MSVC
  -- Found assembler: C:/Program Files (x[8](https://github.com/getsentry/sentry-unity/runs/5077701645?check_suite_focus=true#step:7:8)6)/Microsoft Visual Studio/201[9](https://github.com/getsentry/sentry-unity/runs/5077701645?check_suite_focus=true#step:7:9)/Enterprise/VC/Tools/MSVC/14.29.30[13](https://github.com/getsentry/sentry-unity/runs/5077701645?check_suite_focus=true#step:7:13)3/bin/Hostx64/x64/cl.exe
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  ```

Reply:
Oh that's right, my memory failed me again:

https://github.com/getsentry/sentry-unity/blob/1a2431cd5ee3482bcdd2a92132ac1d2f5a857ab4/Directory.Build.targets#L140-L143

We have Build depend on this step, not the other way around.